File: 48-rotate-image/48-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        1. Transpose of the matrix
        2. Reverse each row i.e list
        """
        for i in range(len(matrix)):
            for j in range(i):
                matrix[i][j],matrix[j][i] = matrix[j][i],matrix[i][j]
        for i in range(len(matrix)):
            matrix[i].reverse()
        
        

File: 48-rotate-image/48-rotate-image.cpp
class Solution {
public:
/*
1. Transpose of the matrix
2. Reverse each row i.e vector<int> 
*/
    void rotate(vector<vector<int>>& matrix) {
        for (int i = 0;i<matrix.size();i++){
            for(int j = 0;j<i;j++){
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
        for (int i = 0;i<matrix.size();i++){
            reverse(matrix[i].begin(),matrix[i].end());
        }
    }
};